### PR TITLE
Fix(ts/analzer) intersectionAndUnionTypes call duplicate error #376

### LINKED ISSUE #376: Reduce errors from assignment failure of unions
https://github.com/dudykr/stc/blob/d50b685dd80aaa0ca4227551bb9cd26af0fd7af4/crates/stc_ts_type_checker/tests/conformance/types/intersection/intersectionAndUnionTypes.ts#L23

should fail with one error, but currently, stc fails with two errors.

```
TS2322

  x context: tried to assign to an element of an intersection type
  | context:
  | lhs = C#6864;
  | Member as {
  |     c: string;
  | };
  | rhs = A#6864;
  | Member as {
  |     a: string;
  | };
  | context: tried to assign a type to an interface
  | context: tried to assign to type elements by converting rhs to a type literal
  | SimpleAssignFailed {
  |     span: Span {
  |         lo: BytePos(
  |             328,
  |         ),
  |         hi: BytePos(
  |             329,
  |         ),
  |         ctxt: #0,
  |     },
  |     cause: Some(
  |         Errors {
  |             span: Span {
  |                 lo: BytePos(
  |                     328,
  |                 ),
  |                 hi: BytePos(
  |                     329,
  |                 ),
  |                 ctxt: #0,
  |             },
  |             errors: [
  |                 MissingFields {
  |                     span: Span {
  |                         lo: BytePos(
  |                             328,
  |                         ),
  |                         hi: BytePos(
  |                             329,
  |                         ),
  |                         ctxt: #0,
  |                     },
  |                     fields: [
  |                         Property(
  |                             PropertySignature {
  |                                 span: Span {
  |                                     lo: BytePos(
  |                                         69,
  |                                     ),
  |                                     hi: BytePos(
  |                                         78,
  |                                     ),
  |                                     ctxt: #0,
  |                                 },
  |                                 accessibility: None,
  |                                 readonly: false,
  |                                 key: Normal {
  |                                     span: Span {
  |                                         lo: BytePos(
  |                                             69,
  |                                         ),
  |                                         hi: BytePos(
  |                                             70,
  |                                         ),
  |                                         ctxt: #0,
  |                                     },
  |                                     sym: Atom('c' type=inline),
  |                                 },
  |                                 optional: false,
  |                                 params: [],
  |                                 type_ann: Some(
  |                                     Keyword(
  |                                         KeywordType {
  |                                             span: Span {
  |                                                 lo: BytePos(
  |                                                     72,
  |                                                 ),
  |                                                 hi: BytePos(
  |                                                     78,
  |                                                 ),
  |                                                 ctxt: #0,
  |                                             },
  |                                             kind: TsStringKeyword,
  |                                             metadata: KeywordTypeMetadata {
  |                                                 common: CommonTypeMetadata {
  |                                                     implicit: false,
  |                                                     infected_by_this_in_object_literal: false,
  |                                                     prevent_converting_to_children: false,
  |                                                     no_expand: false,
  |                                                     ignore_no_expand: false,
  |                                                     contains_infer_type: false,
  |                                                     prevent_complex_simplification: false,
  |                                                     resolved_from_var: false,
  |                                                     prevent_generalization: false,
  |                                                     prevent_tuple_to_array: false,
  |                                                 },
  |                                             },
  |                                         },
  |                                     ),
  |                                 ),
  |                                 type_params: None,
  |                                 metadata: TypeElMetadata {
  |                                     has_default: false,
  |                                 },
  |                                 accessor: Accessor {
  |                                     getter: false,
  |                                     setter: false,
  |                                 },
  |                             },
  |                         ),
  |                     ],
  |                 },
  |             ],
  |         },
  |     ),
  | }
    ,-[$DIR/tests/conformance/types/intersection/intersectionAndUnionTypes.ts:23:1]
 23 | x = aob;
    : ^
    `----
TS2322

  x context: tried to assign to an element of an intersection type
  | context:
  | lhs = D#6864;
  | Member as {
  |     d: string;
  | };
  | rhs = A#6864;
  | Member as {
  |     a: string;
  | };
  | context: tried to assign a type to an interface
  | context: tried to assign to type elements by converting rhs to a type literal
  | SimpleAssignFailed {
  |     span: Span {
  |         lo: BytePos(
  |             328,
  |         ),
  |         hi: BytePos(
  |             329,
  |         ),
  |         ctxt: #0,
  |     },
  |     cause: Some(
  |         Errors {
  |             span: Span {
  |                 lo: BytePos(
  |                     328,
  |                 ),
  |                 hi: BytePos(
  |                     329,
  |                 ),
  |                 ctxt: #0,
  |             },
  |             errors: [
  |                 MissingFields {
  |                     span: Span {
  |                         lo: BytePos(
  |                             328,
  |                         ),
  |                         hi: BytePos(
  |                             329,
  |                         ),
  |                         ctxt: #0,
  |                     },
  |                     fields: [
  |                         Property(
  |                             PropertySignature {
  |                                 span: Span {
  |                                     lo: BytePos(
  |                                         95,
  |                                     ),
  |                                     hi: BytePos(
  |                                         104,
  |                                     ),
  |                                     ctxt: #0,
  |                                 },
  |                                 accessibility: None,
  |                                 readonly: false,
  |                                 key: Normal {
  |                                     span: Span {
  |                                         lo: BytePos(
  |                                             95,
  |                                         ),
  |                                         hi: BytePos(
  |                                             96,
  |                                         ),
  |                                         ctxt: #0,
  |                                     },
  |                                     sym: Atom('d' type=inline),
  |                                 },
  |                                 optional: false,
  |                                 params: [],
  |                                 type_ann: Some(
  |                                     Keyword(
  |                                         KeywordType {
  |                                             span: Span {
  |                                                 lo: BytePos(
  |                                                     98,
  |                                                 ),
  |                                                 hi: BytePos(
  |                                                     104,
  |                                                 ),
  |                                                 ctxt: #0,
  |                                             },
  |                                             kind: TsStringKeyword,
  |                                             metadata: KeywordTypeMetadata {
  |                                                 common: CommonTypeMetadata {
  |                                                     implicit: false,
  |                                                     infected_by_this_in_object_literal: false,
  |                                                     prevent_converting_to_children: false,
  |                                                     no_expand: false,
  |                                                     ignore_no_expand: false,
  |                                                     contains_infer_type: false,
  |                                                     prevent_complex_simplification: false,
  |                                                     resolved_from_var: false,
  |                                                     prevent_generalization: false,
  |                                                     prevent_tuple_to_array: false,
  |                                                 },
  |                                             },
  |                                         },
  |                                     ),
  |                                 ),
  |                                 type_params: None,
  |                                 metadata: TypeElMetadata {
  |                                     has_default: false,
  |                                 },
  |                                 accessor: Accessor {
  |                                     getter: false,
  |                                     setter: false,
  |                                 },
  |                             },
  |                         ),
  |                     ],
  |                 },
  |             ],
  |         },
  |     ),
  | }
    ,-[$DIR/tests/conformance/types/intersection/intersectionAndUnionTypes.ts:23:1]
 23 | x = aob;
    : ^
    `----
```

To match `tsc`, we should merge them

### PR DESCRIPTION
**Pull request by sunrabbit123**
**Description:**

resolve #376

### CODE REVIEW

File: crates/stc_ts_type_checker/tests/conformance/interfaces/interfaceDeclarations/interfaceExtendsObjectIntersectionErrors.ts.stats.rust-debug [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Stats {
     required_error: 14,
     matched_error: 9,
-    extra_error: 1,
```
Comment (sunrabbit123) [inline crates/stc_ts_type_checker/tests/conformance/interfaces/interfaceDeclarations/interfaceExtendsObjectIntersectionErrors.ts.stats.rust-debug:4]: need more investigation

**Review by sunrabbit123 — COMMENTED** [review]
It is a pity that one error has been added

File: crates/stc_ts_type_checker/tests/conformance.pass.txt [hunk 2]
```diff
@@ -1,2500 +1,5000 @@
...
 es6/arrowFunction/emitArrowFunctionWhenUsingArguments07_ES6.ts
+es6/arrowFunction/emitArrowFunctionWhenUsingArguments07_ES6.ts
+es6/arrowFunction/emitArrowFunctionWhenUsingArguments08.ts
 es6/arrowFunction/emitArrowFunctionWhenUsingArguments08.ts
 es6/arrowFunction/emitArrowFunctionWhenUsingArguments08_ES6.ts
+es6/arrowFunction/emitArrowFunctionWhenUsingArguments08_ES6.ts
+es6/arrowFunction/emitArrowFunctionWhenUsingArguments09.ts
 es6/arrowFunction/emitArrowFunctionWhenUsingArguments09.ts
 es6/arrowFunction/emitArrowFunctionWhenUsingArguments09_ES6.ts
+es6/arrowFunction/emitArrowFunctionWhenUsingArguments09_ES6.ts
```
Comment (kdy1) [inline crates/stc_ts_type_checker/tests/conformance.pass.txt:836]: This is duplicated

File: crates/stc_ts_type_checker/tests/conformance.pass.txt [hunk 3]
```diff
@@ -1,2500 +1,5001 @@
 Symbols/ES5SymbolProperty1.ts
+Symbols/ES5SymbolProperty1.ts
```
Comment (sunrabbit123) [inline crates/stc_ts_type_checker/tests/conformance.pass.txt:2]: why duplicate conformance...
I don't know this reason

I use scripts/check.sh, scripts/test.sh, cargo test...

Comment (kdy1) [reply crates/stc_ts_type_checker/tests/conformance.pass.txt:2]: You have to wait for `./scripts/check.sh` to finish

**Review by kdy1 — DISMISSED** [review]
Thanks!

**Review by kdy1 — APPROVED** [review]

**Pull request merged**